code/modules/power/cable.dm

```diff
@@ -1,580 +1,770 @@
-// Cable laying procedures
-//////////////////////////////////////////////
-
-// called when cable_coil is clicked on a turf
-/obj/item/stack/cable_coil/proc/place_turf(turf/T, mob/user, dirnew)
-	if(!isturf(user.loc))
-		return
-
-	if(!isturf(T) || T.intact || !T.can_have_cabling())
-		to_chat(user, "<span class='warning'>You can only lay cables on catwalks and plating!</span>")
-		return
-
-	if(get_amount() < 1) // Out of cable
-		to_chat(user, "<span class='warning'>There is no cable left!</span>")
-		return
-
-	if(get_dist(T,user) > 1) // Too far
-		to_chat(user, "<span class='warning'>You can't lay cable at a place that far away!</span>")
-		return
-
-	for(var/obj/structure/cable/C in T)
-		if(target_type == C.type)
-			to_chat(user, "<span class='warning'>There's already a cable at that position!</span>")
-			return
-
-	var/obj/structure/cable/C = new target_type(T)
-
-	//create a new powernet with the cable, if needed it will be merged later
-	var/datum/powernet/PN = new()
-	PN.add_cable(C)
-
-	for(var/dir_check in GLOB.cardinals)
-		C.mergeConnectedNetworks(dir_check) //merge the powernet with adjacents powernets
-	C.mergeConnectedNetworksOnTurf() //merge the powernet with on turf powernets
-
-	use(1)
-
-	if(C.shock(user, 50))
-		if(prob(50)) //fail
-			new /obj/item/stack/cable_coil(get_turf(C), 1)
-			C.deconstruct()
-
-	return C
-
-/obj/item/stack/cable_coil/five
-	amount = 5
-
-/obj/item/stack/cable_coil/cut
-	amount = null
-	icon_state = "coil2"
-
-/obj/item/stack/cable_coil/cut/Initialize(mapload)
-	. = ..()
-	if(!amount)
-		amount = rand(1,2)
-	pixel_x = rand(-2,2)
-	pixel_y = rand(-2,2)
-	update_icon()
-
-/obj/item/stack/cable_coil/cyborg
-	is_cyborg = 1
-	custom_materials = list()
-	cost = 1
-
-/obj/structure/cable_bridge
-	name = "cable bridge"
-	desc = "A bridge to connect different cable layers, or link terminals to incompatible cable layers."
-	icon = 'icons/obj/power.dmi'
-	icon_state = "cable_bridge"
-	layer = WIRE_LAYER + 0.02 //Above all the cables but below terminals
-	anchored = TRUE
-	obj_flags = CAN_BE_HIT | ON_BLUEPRINTS
-
-/obj/structure/cable_bridge/Initialize(mapload)
-	. = ..()
-	if (!mapload)
-		var/first = TRUE
-		var/datum/powernet/PN
-		for(var/obj/structure/cable/C in get_turf(src))
-			C.update_icon()
-			if(first == TRUE)
-				first = FALSE
-				PN = C.powernet
-				continue
-			propagate_network(C, PN)
-
-/obj/structure/cable_bridge/wirecutter_act(mob/living/user, obj/item/I)
-	. = ..()
-	qdel(src)
-	return TRUE
-
-#undef UNDER_SMES
-#undef UNDER_TERMINAL
+//Use this only for things that aren't a subtype of obj/machinery/power
+//For things that are, override "should_have_node()" on them
+GLOBAL_LIST_INIT(wire_node_generating_types, typecacheof(list(/obj/structure/grille)))
+
+#define UNDER_SMES -1
+#define UNDER_TERMINAL 1
+
+///////////////////////////////
+//CABLE STRUCTURE
+///////////////////////////////
+////////////////////////////////
+// Definitions
+////////////////////////////////
```

Comment:
Could you fix the End-of-Line issues?

Reply:
How to fix it?

Reply:
Do you only do web-edits?
I can guide you through how to fix it using a text-editor if you join the coder channel in discord.

Reply:
I do it after work.

Reply:
Fixed